app/src/main/java/io/homeassistant/companion/android/settings/SettingsActivity.kt

```diff
@@ -74,6 +104,82 @@ class SettingsActivity : BaseActivity() {
         }
     }
 
+    override fun onUserLeaveHint() {
+        super.onUserLeaveHint()
+        runBlocking {
+            integrationUseCase.setAppActive(false)
+        }
+    }
+
+    override fun onPause() {
+        super.onPause()
+        runBlocking {
+            integrationUseCase.setAppActive(false)
+        }
+    }
+
+    override fun onResume() {
+        super.onResume()
+
+        val appLocked = runBlocking {
+            integrationUseCase.isAppLocked()
+        }
+
+        if (appLocked) {
+            authenticating = true
+            authenticator.authenticate(getString(commonR.string.biometric_title))
+            blurView.setBlurEnabled(true)
+        } else {
+            blurView.setBlurEnabled(false)
+        }
```

Comment:
This code is duplicated in `onWindowFocusChanged`. Looking at the `WebViewActivity`, do you need to start authenticating in both functions? Over there you only blur/unblur in `onResume` and handle authenticating in `onWindowFocusChanged`.

If required in both functions, you could extract it to another function and use that in both to remove the duplicate code.

Reply:
Must have pushed the changes too soon. Handling of the authentication in `onResume` and `onWindowFocusChanged` was intended to be similar to how it's done in the webViewActivity. 

After fixing this `onResume` now only blurs, and authentication is handled only in `onWindowFocusChanged`. Just like the webViewActivity does it.